sources/documents/2_RTB/Verbali/Esterni/VerbaleEsterno_2024-05-06/capitolo-riunione.tex

```diff
@@ -0,0 +1,59 @@
+Questo riguarda l'elaborazione della query utente in modo tale da tokenizzarne solo le componenti semantiche ritenute fondamentali, e allo stesso modo l'elaborazione del dizionario dati in formato JSON per renderlo simile al linguaggio naturale, in modo che i modelli, allenati su quest'ultimo, riescano ad estrarne con più facilità il significato.
+
+\textbf{Risposta: }La \glossario{Proponente} ritiene le fasi di pre-processing non necessarie, in quanto il dizionario dati risulta sufficiente per creare degli embeddings efficaci ed associabili alle query utente senza ulteriori alterazioni. 
+Dimostra poi l'utilizzo di txtai per la ricerca semantica tramite BERTino di efederici, un modello operante su lingua italiana, ed invita all'utilizzo delle funzioni di load e save per ridurre l'overhead e gestire con più facilità il testing di più indici.
+
+\textbf{Descrizione:} Il \glossario{Programmatore} espone il problema dello scarso score dei risultati restituiti dalla funzione di search di txtai, e suggerisce la mitigazione del problema tramite un indice intermedio che si frapponga alla query utente e al risultato finale da fornire a una LLM come ChatGPT per ottenere l'effettiva query SQL.
+
+\textbf{Risposta: }La \glossario{Proponente} accoglie l'idea e ne invita la sperimentazione. 
+Riguardo la bassa rilevanza dei documenti restituiti dalla search, suggerisce la sperimentazione di più modelli. 
+Spiega inoltre l'utilizzo da parte di txtai della libreria FAISS, che si occupa della vettorializzazione e indicizzazione oltre che della ricerca semantica, e propone di eseguire dei paragoni con altre librerie interne.
+
+\subsubsection{Esplorazione dei casi d'uso}
+
+\textbf{Descrizione:} L'\glossario{Analista} spiega la struttura dei casi d'uso per le funzioni di debug disponibili per il \glossario{Tecnico}, e in particolare una vista a lui specifica comprendente una lista di tabelle con la corrispondenza tramite il punteggio di score tra la query utente e il risultato ottenuto. 
+Chiede poi se la \glossario{Proponente} avesse pensato a funzionalità aggiuntive per il debugging.
+
+\textbf{Risposta: }La \glossario{Proponente} conferma che la funzione degli strumenti di debug sia mostrare la correttezza del dizionario dati. 
+Propone poi che venga mostrata una spiegazione sull'eventuale esclusione di un risultato che invece sarebbe stato pertinente a favore di uno meno attinente.
+
+\textbf{Descrizione:} L'\glossario{Analista} chiede se sia ragionevole generare un elenco di punteggi, mostrando l'intero \glossario{indice} o gran parte di esso.
+
+\textbf{Risposta: }La \glossario{Proponente} spiega che grazie alla logica di \glossario{back-end} il riscontro sulla query è gerarchico e non avviene ogni volta per calcolo sequenziale, ovvero non si effettua ogni volta il calcolo della distanza del coseno su tutti i vettori dell'indice. 
```

Comment:
\textbf{Risposta:} La \glossario{Proponente} spiega che grazie alla logica di \glossario{back-end}, il processo di ricerca non avviene per calcolo sequenziale, ma attraversa una struttura gerarchica multistrato. Pertanto, la query utente non deve essere confrontata con tutti i vettori dell'indice. 
Eseguire la funzione search non è costoso, mentre lo sarebbe visitare in maniera sequenziale l'intero indice. 
L'unica operazione onerosa, per cui potrebbero essere necessari anche dei minuti, riguarda la creazione dell'indice; per tale motivo sono state introdotte le già menzionate funzioni load e save.